### ISSUE TITLE
Create a rule enforcing colon position for dictionaries

### ISSUE THREAD

**Issue by Noobish1** [realm/SwiftLint#603]
Not sure if this is possible, but i'd love a rule that enforces colon position for dictionaries, I like dictionaries to be defined as `[Key : Value]` with a space on either side of the colon, this rule could be for dictionary instances and also for dictionary type definitions (e.g `[String : AnyObject]`).

**Comment by ceeK** [comment]
Would love this too. 

Personally, I see String: Int as 'string is of type Int' rather than String maps to Int. A separation between the two would be great.

**Comment by phelgo** [comment]
I totally agree we should have a rule for the colon position on dictionaries, but I'm not sure if it should be `[Key : Value]`. I personally prefer `[Key: Value]` which is what the [GitHub code style](https://github.com/github/swift-style-guide) suggests and how Apple itself is writing on [their documentation](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/CollectionTypes.html).

**Comment by ceeK** [comment]
Perhaps then it should be configurable?

**Comment by phelgo** [comment]
@ceeK: That would be ideal! Though it would be cool to agree on a default as well

**Comment by ceeK** [comment]
Yep, and on that point, I think if Apple + Github style guide has it Key: Value, then that should be the default.

**Comment by Noobish1** [comment]
I agree, configurable with key: value as the default.

**Comment by masters3d** [comment]
related: https://github.com/realm/SwiftLint/issues/637

**Comment by lazarevzubov** [comment]
Sorry, but I don't understand, why it was closed.

The `apply_to_dictionaries` parameter of the `colon` rule affects only dictionary contents, but not the dictionary type declaration (as is described in the feature proposal).

To elaborate, assume we have this code:
```swift
let dict: [String : String] = ["string" : "string"]
```

`colon` with `apply_to_dictionaries: true` produces this formatting:
```swift
let dict: [String: String] = ["string": "string"]
```

`colon` with `apply_to_dictionaries: false` produces this:
```swift
let dict: [String: String] = ["string" : "string"]
```

Whereas the feature proposal requires an opportunity to produce this:
```swift
let dict: [String : String] = ["string" : "string"]
```

Or did I miss something?

**Comment by xSiavAx** [comment]
+1 
Would be nice to have such rule

**Comment by SimplyDanny** [comment]
The format requested by this issue is indeed not covered by the existing `colon` rule.
